src/Sarif/Writers/MimeType.cs

```diff
@@ -27,43 +30,22 @@ public static string DetermineFromFileExtension(string path)
                     // Each entry in the table is of the form [ mimeType, ext1, ext2, ... extN ]
                     for (int idx = 1; idx < tableEntry.Length; ++idx)
                     {
-                        if (HasExtension(path, tableEntry[idx]))
+                        if (HasExtension(fileUri.AbsolutePath, tableEntry[idx]))
                         {
                             return tableEntry[0];
                         }
                     }
                 }
             }
 
-            if (System.IO.Directory.Exists(path))
+            if(fileUri.IsAbsoluteUri && fileUri.IsFile && System.IO.Directory.Exists(fileUri.LocalPath))
             {
                 return MimeType.Directory;
             }
 
             return MimeType.Binary;
         }
 
-        /// <summary>Guesses an appropriate MIME type given the extension from a file name.</summary>
-        /// <param name="fileName">File name from which MIME type shall be guessed.</param>
-        /// <returns>A string corresponding to the likely MIME type of <paramref name="fileName"/> given
-        /// its extension.</returns>
-        public static string DetermineFromFileExtension(Uri fileUri)
-        {
-            if (fileUri == null)
-            {
-                throw new ArgumentNullException(nameof(fileUri));
-            }
-
-            string fileName = fileUri.ToString();
-
```

Comment:
So, this issue is a well-known weakness in our API, our handling of URIs is very inconsistent across schema, relative vs. absolute, etc. 

The issue with the change that you've proposed is that I'm not sure it is resilient to all kinds of inputs (such as relative paths). I see that we only have one upstream caller to this API in Artifact. Create, which looks like it is over-aggressively generating the mime type. That caller is also a place we could consider a fix (unless we are convinced this API is useful in general, which I am not)

Try this, create a new branch, and for this line I've highlighted:

    string fileName = fileUri.ToString().

See if this line succeeds:

    string fileName = fileUri.GetFileName();

This helper is supposed to elide all differences in URL kinds and retrieve what is effectively the file name.

This helper lives in Sarif/ExtensionMethods.cs. I'm guessing this one line change may fix the issue... But looking forward to whether you discover if this is so.

If you've spent enough time on this break and would prefer we finish the investigation let me know. :) Thanks for your help so far, I appreciate it.